pkg/network/protocols/http/monitor_test.go

```diff
@@ -262,44 +307,82 @@ func TestHTTPMonitorIntegrationWithNAT(t *testing.T) {
 
 	targetAddr := "2.2.2.2:8080"
 	serverAddr := "1.1.1.1:8080"
-	t.Run("with keep-alives", func(t *testing.T) {
-		testHTTPMonitor(t, targetAddr, serverAddr, 100, testutil.Options{
-			EnableKeepAlives: true,
-		})
-	})
-	t.Run("without keep-alives", func(t *testing.T) {
-		testHTTPMonitor(t, targetAddr, serverAddr, 100, testutil.Options{
-			EnableKeepAlives: false,
+
+	for _, TCPTimestamp := range []struct {
+		name  string
+		value bool
+	}{
+		{name: "without TCP timestamp option", value: false},
+		{name: "with TCP timestamp option", value: true},
+	} {
+		t.Run(TCPTimestamp.name, func(t *testing.T) {
+			t.Run("with keep-alives", func(t *testing.T) {
+				testHTTPMonitor(t, targetAddr, serverAddr, 100, testutil.Options{
+					EnableKeepAlives:   true,
+					EnableTCPTimestamp: TCPTimestamp.value,
+				})
+			})
+			t.Run("without keep-alives", func(t *testing.T) {
+				testHTTPMonitor(t, targetAddr, serverAddr, 100, testutil.Options{
+					EnableKeepAlives:   false,
+					EnableTCPTimestamp: TCPTimestamp.value,
+				})
+			})
 		})
-	})
+	}
 }
 
 func TestUnknownMethodRegression(t *testing.T) {
-	monitor := newHTTPMonitor(t)
+	for _, TCPTimestamp := range []struct {
+		name  string
+		value bool
+	}{
+		{name: "without TCP timestamp option", value: false},
+		{name: "with TCP timestamp option", value: true},
+	} {
 
-	// SetupDNAT sets up a NAT translation from 2.2.2.2 to 1.1.1.1
-	netlink.SetupDNAT(t)
+		t.Run(TCPTimestamp.name, func(t *testing.T) {
+			targetAddr := "2.2.2.2:8080"
+			serverAddr := "1.1.1.1:8080"
+			srvDoneFn := testutil.HTTPServer(t, serverAddr, testutil.Options{
+				EnableTLS:          false,
+				EnableKeepAlives:   true,
+				EnableTCPTimestamp: TCPTimestamp.value,
+			})
+			defer srvDoneFn()
 
-	targetAddr := "2.2.2.2:8080"
-	serverAddr := "1.1.1.1:8080"
-	srvDoneFn := testutil.HTTPServer(t, serverAddr, testutil.Options{
-		EnableTLS:        false,
-		EnableKeepAlives: true,
-	})
-	defer srvDoneFn()
+			monitor := newHTTPMonitor(t)
+			// SetupDNAT sets up a NAT translation from 2.2.2.2 to 1.1.1.1
+			netlink.SetupDNAT(t)
 
-	requestFn := requestGenerator(t, targetAddr, emptyBody)
-	for i := 0; i < 100; i++ {
-		requestFn()
-	}
+			requestFn := requestGenerator(t, targetAddr, emptyBody)
+			for i := 0; i < 100; i++ {
+				requestFn()
+			}
 
-	time.Sleep(10 * time.Millisecond)
-	stats := monitor.GetHTTPStats()
+			time.Sleep(2 * time.Second)
+			stats := monitor.GetHTTPStats()
 
-	for key := range stats {
-		if key.Method == MethodUnknown {
-			t.Error("detected HTTP request with method unknown")
-		}
+			nonDstPortRequests := int64(0)
+			for key := range stats {
+				if key.DstPort != 8080 {
+					nonDstPortRequests++
+				}
```

Comment:
why is it useful?

Reply:
yes in the CI, some os have some checker running with http and I added a check on numbers of requests that we generate/received